### PR TITLE
Prof: User views success message on auth success

### PR DESCRIPTION
**Pull request by chidioguejiofor**
#### What does this PR 
- adds a navigation state that controls navigation from one page
to the next
- creates a PageWrapper component that retrieves data from the
connected-react-router to the navigation state for all pages in the app.
- adds sagas for navigation state
- adds reducers for navigation state

#### Any background context?
- When a page loads, the info stored in `connected-react-router` is retrieved and stored in the `navigation` state. Any page that wants to use data from previous pages would have to connect to `navigation` state

#### Manual Testing Steps
- Sign up a user via the index route
- On success, you should be redirected to the Auth success page and `userData` stored in the `navigation` state
- When you refresh the page and check the state, you reload the state you should still see `userData` in navigation state

#### Screenshots:
Screenshot 2019-11-29 at 9 24 00 AM
Screenshot 2019-11-29 at 9 37 46 AM
Screenshot 2019-11-29 at 9 37 18 AM
Screenshot 2019-11-29 at 9 36 04 AM

#### Relevant Trello Tickets
- [Auth Success](https://trello.com/c/jawHAM1o/37-implement-authentication-success-feedback)

### CODE REVIEW

**Comment by chukwuemekachm** [comment]
Nice on @chidioguejiofor, please look at this screenshot, the font colour and text should match what we have below. on the terms and condition.

Screenshot 2019-11-29 at 10 22 54 AM

**Review by chukwuemekachm — CHANGES_REQUESTED** [review]
Nice work @chidioguejiofor.

Please see my comments below.

File: src/components/layouts/EmailSentSuccess/index.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+function EmailSentSuccess({navigationData}) {
+    const name = navigationData.userData? navigationData.userData.firstName: 'there!'
+    const SIGNUP_MESSAGE =  `Hi ${name} your signup was  successful and a verification
+        mail was sent to your email. Please confirm your email to continue`;
+     const RESET_PASSWORD_MESSAGE =
+         `Hi there! Instructions to reset your password has been sent to your email.
+         Please check it out to reset your password thanks!`;
+  return (
+    <EmailSentSuccess.Wrapper>
```
Comment (chukwuemekachm) [inline src/components/layouts/EmailSentSuccess/index.tsx]: This should be a component in the`UI` folder for now and not a layout. It's page uses the `AuthenticationLayout`.

Comment (chidioguejiofor) [reply src/components/layouts/EmailSentSuccess/index.tsx]: done. Also decided not to connect `EmailSentSuccess` to store. Instead sent down props from `AuthFeedback`

**Comment by chidioguejiofor** [comment]
> Nice on @chidioguejiofor, please look at this screenshot, the font colour and text should match what we have below. on the terms and condition.
> 
> Screenshot 2019-11-29 at 10 22 54 AM

I have updated this to 

Screenshot 2019-11-29 at 1 09 42 PM

**Comment by chidioguejiofor** [comment]
I think we also need to move that `Terms and Conditions` to the `SignupForm`

**Pull request merged**